### PR TITLE
feat: base files for static preview

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Create basic outline of controllers/services/routes for static preview.

Next PR will introduce the preview config validation

### CODE REVIEW

File: packages/core/content-manager/server/src/preview/controllers/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import type { Plugin } from '@strapi/types';
+import { createPreviewController } from './preview';
+
+export const controllers = {
+  preview: createPreviewController,
+} as unknown as Plugin.LoadedPlugin['controllers'];
```
Comment (remidej) [inline packages/core/content-manager/server/src/preview/controllers/index.ts:6]: Why is unknown needed?

and I thought you would need to call the function, I didn't know this style was allowed

File: packages/core/content-manager/server/src/preview/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import type { Plugin } from '@strapi/types';
+import { previewRouter } from './preview';
+
+/**
+ * The routes will me merged with the other Content Manager routers,
```
Comment (remidej) [inline packages/core/content-manager/server/src/preview/routes/index.ts:5]:
```suggestion
 * The routes will be merged with the other Content Manager routers,
```

File: packages/core/content-manager/server/src/preview/controllers/index.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/content-manager/server/src/preview/controllers/index.ts:6]: apparently the controller types do not account for using the factory itself, but it works (copied it from content history)

I will add a comment!

File: packages/core/content-manager/server/src/preview/index.ts [hunk 3]
```diff
@@ -1,21 +1,38 @@
...
   return {
     bootstrap() {
       // eslint-disable-next-line no-console -- TODO remove when we have real functionality
       console.log('Bootstrapping preview server');
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- TODO: Remove when implemented
+      const config = strapi.config.get('admin.preview');
+
+      // TODO: Validation
+      // TODO: Disable feature if config is not valid
```
Comment (remidej) [inline packages/core/content-manager/server/src/preview/index.ts:31]: Maybe it's easier to just crash the app?

**Review by remidej — APPROVED** [review]

File: packages/core/content-manager/server/src/preview/index.ts (hunk 3 above)
Comment (Marc-Roig) [reply packages/core/content-manager/server/src/preview/index.ts:31]: totally! I will do that in the [next](https://github.com/strapi/strapi/pull/21561) pr

**Pull request merged**